### PR TITLE
[hooks] Add light client

### PR DESCRIPTION
**Pull request by nikitayutanov**
Expand `useApi` hook to support light client connection and smooth network change without forced page reload.

`api` is more type safe now - it's `undefined` before initialization. Use `isApiReady` variable to validate it's current state.

### CODE REVIEW

File: utils/gear-hooks/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@gear-js/react-hooks",
-  "version": "0.6.7",
+  "version": "0.8.0",
```
Comment (ereburg) [inline utils/gear-hooks/package.json:3]: where is 0.7 version?

File: utils/gear-hooks/src/context/Api.tsx [hunk 2]
```diff
@@ -1,33 +1,90 @@
...
-  isApiReady: boolean;
+type WsProviderArgs = {
+  endpoint: string | string[];
+  autoConnectMs?: number | false;
+  headers?: Record<string, string>;
+  timeout?: number;
+};
+
+type ScProviderArgs = {
+  spec: string | Sc.WellKnownChain;
```
Comment (osipov-mit) [inline utils/gear-hooks/src/context/Api.tsx:16]: Is wellknownchain necessary here?

**Review by ereburg — APPROVED** [review]

File: utils/gear-hooks/package.json (hunk 1 above)
Comment (nikitayutanov) [reply utils/gear-hooks/package.json:3]: 👍, pulled changes from `main`

**Review by faizov — APPROVED** [review]

File: utils/gear-hooks/src/context/Api.tsx (hunk 2 above)
Comment (nikitayutanov) [reply utils/gear-hooks/src/context/Api.tsx:16]: Absolutely not, yeah, I guess it's misleading since we only need Gear endpoints.

**Review by YauheniDraichykau — APPROVED** [review]

**Review by osipov-mit — APPROVED** [review]

**Pull request merged**